deploy/databases/linux/mysql/roles/onafterstart/tasks/main.yml

```diff
@@ -0,0 +1,74 @@
+---
+
+- name: Ensure newrelic_license_key is defined
+  fail:
+    msg: "newrelic_license_key is required for configuring newrelic infrastructure agent"
+  when: newrelic_license_key is not defined
+
+- name: Ensure database_port is defined
+  fail:
+    msg: "database_port is required for configuring mysql integration"
+  when: database_port is not defined
+
+- name: Ensure database_password is defined
+  fail:
+    msg: "database_password is required for configuring mysql integration"
+  when: database_password is not defined
+
+- name: Check if infrastructure agent logs exist
+  stat:
+    path: /var/log/nr-infra.log
+  register: stat_result
+
+# Clear logs (from previous deployments) to avoid a false positive in the verification step.
+- name: Clear agent logs if they exist
+  shell: truncate -s 0 /var/log/nr-infra.log
+  when: stat_result.stat.exists
+  become: true
+
+- name: Run NewRelic Infrastructure Agent role
+  include_role:
+    name: newrelic.newrelic-infra
+    apply:
+      become: true
+  vars:
+    nrinfragent_config:
+      license_key: "{{ newrelic_license_key }}"
+      log_file: /var/log/nr-infra.log
+      log_to_stdout: false
+    nrinfragent_integrations:
+      - { name: nri-mysql, state: "latest" }
+
+- name: Create db user for newrelic mysql integration
+  mysql_user:
+    login_user: "root"
+    login_password: "{{ database_password }}"
+    name: "newrelic"
+    password: "{{ database_password }}"
+    state: present
+    host: "localhost"
+    priv: "*.*:GRANT,REPLICATION CLIENT"
```

Comment:
I believe it also needs the SELECT permission, despite what our NR documentation says (I have a feedback item to the infra OHI team about that).

Reply:
do you know why this is needed? the integration appears to be working

Reply:
On some version of linux/mysql, this was required. I believed Debian10 with latest mysql version